flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/CreditBasedPartitionRequestClientHandler.java

```diff
@@ -249,7 +248,7 @@ private void decodeMsg(Object msg) throws Throwable {
 			NettyMessage.BufferResponse bufferOrEvent = (NettyMessage.BufferResponse) msg;
 
 			RemoteInputChannel inputChannel = inputChannels.get(bufferOrEvent.receiverId);
-			if (inputChannel == null) {
+			if (inputChannel == null || inputChannel.isReleased()) {
```

Comment:
we should consider the case of null data buffer to also cancel request.

Reply:
After some offline discussion, we come to the consistency that the current implementation should be able to cover the above case.